desc/compute/_stability.py

```diff
@@ -231,3 +231,365 @@ def _magnetic_well(params, transforms, profiles, data, **kwargs):
         0,  # coefficient of limit is V_r / V_rr, rest is finite
     )
     return data
+
+
+@register_compute_fun(
+    name="ideal ball gamma2",
+    label="\\gamma^2",
+    units=" ",
+    units_long=" ",
+    description="ideal ballooning growth rate" + "requires data along a field line",
+    dim=1,
+    params=["Psi"],
+    transforms={"grid": []},
+    profiles=[],
+    coordinates="rtz",
+    data=[
+        "a",
+        "g^aa",
+        "g^ra",
+        "g^rr",
+        "cvdrift",
+        "cvdrift0",
+        "|B|",
+        "B^zeta",
+        "p_r",
+        "iota",
+        "shear",
+        "psi",
+        "psi_r",
+        "rho",
+    ],
+    source_grid_requirement={"coordinates": "raz", "is_meshgrid": True},
+)
+def _ideal_ballooning_gamma2(params, transforms, profiles, data, **kwargs):
+    """
+    Ideal-ballooning growth rate finder.
+
+    This function uses a finite-difference method
+    to calculate the maximum growth rate against the
+    infinite-n ideal ballooning mode. The equation being solved is
+
+    d/dζ(g dX/dζ) + c X = lam f X, g, f > 0
+
+    where
+
+    𝛋 = 𝐛 ⋅∇ 𝐛
+    g = a_N^3 * B_N * (b ⋅∇ζ) * |∇α|², / B,
+    c = a_N/B_N * (1/ b ⋅∇ζ) * dψ/dρ * dp/dψ * (b × 𝛋) ⋅|∇α|/ B**2,
+    f = a_N * B_N^3 *|∇α|² / B^3 * (1/ b ⋅∇ζ) ,
+    are needed along a field line to solve the ballooning equation once.
+
+    To obtain the parameters g, c, and f, we need a set of parameters
+    provided in the list ``data`` above. Here's a description of
+    these parameters:
+
+    - a: minor radius of the device
+    - g^aa: |grad alpha|^2, field line bending term
+    - g^ra: (grad alpha dot grad rho) integrated local shear
+    - g^rr: |grad rho|^2 flux expansion term
+    - cvdrift: geometric factor of the curvature drift
+    - cvdrift0: geoetric factor of curvature drift 2
+    - |B|: magnitude of the magnetic field
+    - B^zeta: inverse of the jacobian
+    - p_r: dp/drho, pressure gradient
+    - phi: coordinate describing the position in the toroidal angle
+    along a field line
+    """
+    source_grid = transforms["grid"].source_grid
+    # Vectorize in rho later
+    rho = data["rho"]
+
+    psi_b = params["Psi"] / (2 * jnp.pi)
+    a_N = data["a"]
+    B_N = 2 * psi_b / a_N**2
+
+    N_zeta0 = int(15)
+    # up-down symmetric equilibria only
+    zeta0 = jnp.linspace(-0.5 * jnp.pi, 0.5 * jnp.pi, N_zeta0)
+
+    iota = data["iota"]
+    shear = data["shear"]
+    psi = data["psi"]
+    sign_psi = jnp.sign(psi)
+    sign_iota = jnp.sign(iota)
+
+    phi = source_grid.nodes[:, 2]
+
+    N_alpha = int(source_grid.num_alpha)
+    N_zeta = int(len(phi) / N_alpha)
+
+    B = jnp.reshape(data["|B|"], (N_alpha, 1, N_zeta))
```

Comment:
I think you might have to change this after #1214 was merged, you can use
`B = transforms['grid'].meshgrid_reshape(data['|B|'], "arz")`, similar for the other places you reshape stuff to separate the fieldlines. I can do it tomorrow, let's see if the tests fail.